spec/forms/backup_code_verification_form_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.describe BackupCodeVerificationForm do
+  let(:form) { described_class.new(user) }
+  let(:user) { create(:user, :fully_registered) }
+  let(:backup_codes) do
+    BackupCodeGenerator.new(user).create
+  end
+  let(:params) do
+    {
+      backup_code: code,
+    }
+  end
+  subject(:result) { form.submit(params) }
+
+  describe '#submit' do
+    context 'with a valid backup code' do
+      let(:code) { backup_codes.first }
+      let(:backup_code_config) do
+        BackupCodeConfiguration.find_with_code(code: code, user_id: user.id)
+      end
+
+      it 'returns succcess' do
+        expect(result.success?).to eq(true)
+        expect(result.extra[:multi_factor_auth_method]).to eq('backup_code')
+        expect(result.extra[:multi_factor_auth_method_created_at]).
+          to eq(backup_code_config.created_at)
+      end
+
+      it 'marks code as used' do
+        subject
+
+        expect(backup_code_config.reload.used_at).not_to eq(nil)
```

Comment:
We could consider using an [RSpec change matcher](https://rubydoc.info/gems/rspec-expectations/RSpec%2FMatchers:change) to check that it flips between the two expected states as a result of the call.

```suggestion
        expect { result }.to change { backup_code_config.reload.used_at }.from(nil).to kind_of(Time)
```